languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -20273,7 +20273,66 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+             inflections. I mean, 'dever' -> 'deve' is pretty straightforward, but 'dever' -> 'devo' less so.  -->
+        <rulegroup id="DOUBLE_INFINITIVE" name="Infinitivo duplo">
+            <short>Possível erro de digitação.</short>
+            <!-- This contains the original structure as described in premium issue 5538
+                 All sentences herein are of the shape: 'dever ter feito', and as such will usually indicate
+                 some kind of past tense epistemic modality, for which the double infinitive is very probably ungrammatical. -->
+            <rule default="temp_off">
+                <!-- This antipattern refers to constructions of the type 'the fact that...', for which
+                     a personal (i.e. inflected) infinitive can be used. -->
+                <antipattern>
+                    <token regexp="yes">fac?to|questão|possibilidade|hipótese|apesar</token>
+                    <token regexp="yes" skip="-1" postag_regexp="yes" postag="SPS00(:.+)?">
+                        d([eoa]|es[st][ea]|aquel[ea]|uma?|is[st]o)
+                        <exception postag_regexp="yes" postag="V.+"/>
+                    </token>
+                    <token postag_regexp="yes" postag="VMN.+">dever</token>
+                    <token postag_regexp="yes" postag="R." min="0"/>
+                    <token>ter</token>
+                    <token postag_regexp="yes" postag="V.P.+"/>
+                    <example>A questão de o segredo dever ter sido revelado não nos afeta.</example>
+                    <example>O fato do contrato já dever ter sido assinado é irrelevante.</example>
+                </antipattern>
+                <antipattern>
+                    <token postag="CS"/>
+                    <token postag="CS" skip="-1">
+                        de
+                        <exception postag_regexp="yes" postag="V.+"/>
+                    </token>
+                    <token postag_regexp="yes" postag="VMN.+">dever</token>
+                    <token postag_regexp="yes" postag="R." min="0"/>
+                    <token>ter</token>
+                    <token postag_regexp="yes" postag="V.P.+"/>
+                    <example>Apesar de ele dever já ter entrado, tranquei a porta.</example>
+                </antipattern>
+                <!-- A pesky little case, where 'a' + inf. actually generally fits..? -->
+                <antipattern>
+                    <token postag="SPS00">a</token>
+                    <token postag_regexp="yes" postag="VMN.+">dever</token>
+                    <token postag_regexp="yes" postag="R." min="0"/>
+                    <token>ter</token>
+                    <token postag_regexp="yes" postag="V.P.+"/>
+                    <example>A cidade ficou-lhe a dever não ter sido invadida.</example>
+                </antipattern>
+                <pattern>
+                    <token postag_regexp="yes" postag="SENT_START|[DNPARZ].+" min="1"/>
```

Comment:
`min="1"` is not really necessary here

Reply:
It __seems__ to work as intended, i.e. at least one and at most infinity of tokens tagged with that regex. AFAIU the `skip` attribute would better fit the **next** token – but I think when I tried that it didn't quite do what I wanted? 😠

Reply:
Did you test `min="1" max="-1"` instead of `min="1"`? If it works, let's see what we get.

Reply:
I did test it, yes, but now the more I think about it the less sense it makes. Maybe you're right that matching a single one of those tokens should be enough for a healthy match. Sehr verwirrend.

Reply:
Two possibilities here:

**1.** add `skip="-1"` and see what happens (and if we'd miss any matches without it)
**2.** remove all the skips in the first token altogether (or limit them to just a few tokens) and just go with that for now

We could also try 1 and then change to 2 if results get too crazy.

Reply:
Occam's etc., removed the silly min/max in the new commit. As far as I can tell we don't lose any coverage and don't require any new antipatterns.